src/SFML/Graphics/Shader.cpp

```diff
@@ -102,8 +110,16 @@ sfShader* sfShader_createFromMemory(const char* vertexShader, const char* geomet
         }
         else
         {
-            // vertex + geometry + fragment shaders
-            success = shader->This.loadFromMemory(vertexShader, geometryShader, fragmentShader);
+            if (!vertexShader && !fragmentShader)
+            {
+                // geometry shader only
+                success = shader->This.loadFromMemory(geometryShader, sf::Shader::Geometry);
+            }
+            else
```

Comment:
What about the situation where either vertexShaderFilename or fragmentShaderFilename is not set, i.e. shouldn't there be three cases as above?